client.go

```diff
@@ -31,321 +37,131 @@ func NewClient(apiKey string, options ...Option) *Client {
+// Returns an array of segments that the given identity is part of
+func (c *Client) GeIdentitySegments(identifier string, traits []*Trait) ([]*segments.SegmentModel, error) {
+	if env, ok := c.environment.Load().(*environments.EnvironmentModel); ok {
+		identity := buildIdentityModel(identifier, env.APIKey, traits)
+		return flagengine.GetIdentitySegments(env, &identity), nil
 	}
-	flag := findFeatureFlag(flags, name)
-	return flag != nil && flag.Enabled, nil
-}
-
-// GetValue returns value of given feature (remote config)
-//
-// Returned value can have one of following types: bool, int, string
-func (c *Client) GetValue(name string) (interface{}, error) {
-	return c.GetValueWithContext(context.TODO(), name)
-}
-
-// GetValueWithContext returns value of given feature (remote config)
-//
-// Returned value can have one of following types: bool, int, string
-func (c *Client) GetValueWithContext(ctx context.Context, name string) (interface{}, error) {
-	flags, err := c.GetFeaturesWithContext(ctx)
-	if err != nil {
-		return "", err
-	}
-	flag := findFeatureFlag(flags, name)
-	if flag != nil {
-		return convertValue(flag.StateValue), nil
-	}
-
-	return "", fmt.Errorf("feature flag '%s' not found", name)
-}
+	return nil, errors.New("Local evaluation required to obtain identity segments")
 
-// GetUserValue return value of given feature (remote config) as defined for given user
-//
-// Returned value can have one of following types: bool, int, string
-func (c *Client) GetUserValue(user User, name string) (interface{}, error) {
-	return c.GetUserValueWithContext(context.TODO(), user, name)
 }
-
-// GetUserValueWithContext return value of given feature (remote config) as defined for given user
-//
-// Returned value can have one of following types: bool, int, string
-func (c *Client) GetUserValueWithContext(ctx context.Context, user User, name string) (interface{}, error) {
-	flags, err := c.GetUserFeaturesWithContext(ctx, user)
+func (c *Client) GetEnvironmentFlagsFromAPI(ctx context.Context) (Flags, error) {
+	resp, err := c.client.NewRequest().
+		SetContext(ctx).
+		Get(c.config.baseURL + "flags/")
 	if err != nil {
-		return "", err
-	}
-	flag := findFeatureFlag(flags, name)
-	if flag != nil {
-		return convertValue(flag.StateValue), nil
+		return Flags{}, err
 	}
-	return "", fmt.Errorf("feature flag '%s' not found", name)
-}
-
-// GetTrait returns trait defined for given user
-func (c *Client) GetTrait(user User, key string) (*Trait, error) {
-	return c.GetTraitWithContext(context.TODO(), user, key)
-}
-
-// GetTraitWithContext returns trait defined for given user
-func (c *Client) GetTraitWithContext(ctx context.Context, user User, key string) (*Trait, error) {
-	traits, err := c.GetTraitsWithContext(ctx, user, key)
-	if err != nil {
-		return nil, err
-	} else if len(traits) == 0 {
-		return nil, fmt.Errorf("trait '%s' not found", key)
+	if !resp.IsSuccess() {
+		return Flags{}, errors.New("Unable to get valid response from Flagsmith API.")
```

Comment:
Do we have custom errors defined somewhere?

Reply:
Nope

Reply:
Can we?